xapian-core/include/xapian/queryparser.h

```diff
@@ -234,6 +234,79 @@ class XAPIAN_VISIBILITY_DEFAULT RangeProcessor
     }
 };
 
+// Handles a file size range
+
+class XAPIAN_VISIBILITY_DEFAULT FileSizeRangeProcessor : public RangeProcessor {
+  public:
+    /** Constructor.
+     *
+     *  @param slot_    The value slot number to query.
+     *
+     *  @param str_     A string to look for to recognise values as belonging
+     *                  to this file-size range.
+     *
+     *  @param flags_   Zero or more of the following flags, combined with
+     *          bitwise-or:
+     *           * Xapian::RP_SUFFIX - require @a str_ as a suffix
+     *             instead of a prefix.
+     *           * Xapian::RP_REPEATED - optionally allow @a str_
+     *             on both ends of the range - e.g. $1..$10 or
+     *             5m..50m.  By default a prefix is only checked for on
+     *             the start (e.g. date:1/1/1980..31/12/1989), and a
+     *             suffix only on the end (e.g. 2..12kg).
+     *
+     *  The string supplied in str_ is used by @a operator() to decide whether
+     *  the pair of strings supplied to it constitute a valid range.  If
+     *  prefix_ is true, the first value in a range must begin with str_ (and
+     *  the second value may optionally begin with str_);
+     *  if prefix_ is false, the second value in a range must end with str_
+     *  (and the first value may optionally end with str_).
+     *
+     *  If str_ is empty, the setting of prefix_ is irrelevant, and no special
+     *  strings are required at the start or end of the strings defining the
+     *  range.
+     *
+     *  The remainder of both strings defining the endpoints must be valid
+     *  floating point numbers.
+     *
+     *  FileSizeRangeProcessor will handle file size as:
+     *  B - Bytes
+     *  K - Kilo Bytes
+     *  M - Mega Bytes
+     *  G - Giga Bytes
+     *  For example, if str_ is "size:",  Xapian::RP_SUFFIX is not specified,
+     *  and the range processor has been added to the queryparser, the
+     *  queryparser will accept "size:10K..1M".
+     *  Examples of valid ranges
+     *    size:10K..1M i.e. 10 KB to 1 MB
+     *  Examples of invalid ranges
+     *    size:10K..20
+     *    size:10..20
```

Comment:
If it were in `KB` that would provide poor backwards compatibility as the meaning of such ranges would change by ~3 orders of magnitude.

Bytes seems most logical anyway.